terraform/e2e/eks/main.tf

```diff
@@ -0,0 +1,155 @@
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Principal": {
+        "Service": "ec2.amazonaws.com"
+      },
+      "Action": "sts:AssumeRole"
+    }
+  ]
+}
+POLICY
+}
+
+resource "aws_iam_role_policy_attachment" "node_AmazonEKSWorkerNodePolicy" {
+  policy_arn = "arn:aws:iam::aws:policy/AmazonEKSWorkerNodePolicy"
+  role       = aws_iam_role.node_role.name
+}
+
+resource "aws_iam_role_policy_attachment" "node_AmazonEKS_CNI_Policy" {
+  policy_arn = "arn:aws:iam::aws:policy/AmazonEKS_CNI_Policy"
+  role       = aws_iam_role.node_role.name
+}
+
+resource "aws_iam_role_policy_attachment" "node_AmazonEC2ContainerRegistryReadOnly" {
+  policy_arn = "arn:aws:iam::aws:policy/AmazonEC2ContainerRegistryReadOnly"
+  role       = aws_iam_role.node_role.name
+}
+
+resource "aws_iam_role_policy_attachment" "node_CloudWatchAgentServerPolicy" {
+  policy_arn = "arn:aws:iam::aws:policy/CloudWatchAgentServerPolicy"
+  role       = aws_iam_role.node_role.name
+}
+
+resource "null_resource" "kubectl" {
+  depends_on = [
+    aws_eks_cluster.this,
+    aws_eks_node_group.this
+  ]
+  provisioner "local-exec" {
+    command = <<-EOT
+      ${local.aws_eks} update-kubeconfig --name ${aws_eks_cluster.this.name}
+      ${local.aws_eks} list-clusters --output text
+      ${local.aws_eks} describe-cluster --name ${aws_eks_cluster.this.name} --output text
+    EOT
+  }
+}
+
+resource "null_resource" "install_helm_release_and_sample_app" {
+  depends_on = [
+    null_resource.kubectl,
+    aws_eks_cluster.this,
+    aws_eks_node_group.this
+  ]
+
+  provisioner "local-exec" {
+    command = <<-EOT
+      ${local.aws_eks} update-kubeconfig --name ${aws_eks_cluster.this.name}
+
+      helm upgrade --install amazon-cloudwatch-observability \
+        ${path.module}/helm-charts/charts/amazon-cloudwatch-observability \
+        --values ${path.module}/helm-charts/charts/amazon-cloudwatch-observability/values.yaml \
+        --set clusterName=${aws_eks_cluster.this.name} \
+        --set region=${var.region} \
+        --set agent.image.repository=${var.cloudwatch_agent_repository} \
+        --set agent.image.tag=${var.cloudwatch_agent_tag} \
+        --set agent.image.repositoryDomainMap.public=${var.cloudwatch_agent_repository_url} \
+        --set manager.image.repository=${var.cloudwatch_agent_operator_repository} \
+        --set manager.image.tag=${var.cloudwatch_agent_operator_tag} \
+        --set manager.image.repositoryDomainMap.public=${var.cloudwatch_agent_operator_repository_url} \
+        --namespace amazon-cloudwatch \
+        --create-namespace \
+        ${var.agent-config != "" ? "--set-json agent.config='${jsonencode(jsondecode(file(var.agent-config)))}'" : ""} \
+        ${var.otel-config != "" ? "--set-file agent.otelConfig=${yamlencode(yamldecode(file(var.otel-config)))}" : ""} \
+        ${var.prometheus-config != "" ? "--set-file agent.prometheus.config=${yamlencode(yamldecode(file(var.prometheus-config)))}" : ""}
```

Comment:
I think all these should be pulled from values.yaml located in test directory.  When we do it manually like this we get less control over tests and it decreases our agility in updating our test framework.

Reply:
That's fair, but there's an issue with that approach in case we update `values.yaml` in the future. Setting it like this ensures to target values we will always maintain.